lib/clients/websocket.js

```diff
@@ -41,11 +41,10 @@ class WebsocketClient extends EventEmitter {
   }
 
   disconnect() {
-    if (!this.socket) {
+    if (!this.socket || this.socket.readyState !== Websocket.OPEN) {
```

Comment:
Should also consider `CONNECTING`

Reply:
The CONNECTING was omitted on purpose. The meaning here might be that someone wants to abort the connection. In this case the socket will abort and throw an exception 'WebSocket was closed before the connection was established'. If this is the behaviour we want, I can fix it.

Reply:
Interesting. That error might still be more descriptive than the current message, but I'm fine with omitting it.